### PR TITLE
Feat: drop ruby 2.3

### LINKED ISSUE #7453: Remove support for Ruby 2.3.x
## Summary

Remove explicit support for Ruby 2.3. If it continues to work as part of the 4.x release series, that's great. When Ruby 2.3 stops working with the 4.x series, then we've got a policy in place that says we don't support it.

## Motivation

Ruby 2.3 will be end of life on 2019-03-31. I want to be ahead of the game and stop supporting a version that won't be supported upstream earlier than upstream drops support for it.

### PR DESCRIPTION
**Pull request by DirtyF**
This is a 🙋 feature or enhancement.

## Summary

Support only current active Ruby versions (2.4, 2.5, 2.6)

## Context

Fix #7453 

## Semver Changes

Major (4.0)

### CODE REVIEW

**Comment by ashmaroli** [comment]
:warning: JRuby-9.2.5.0 doesn't work with refinements..

**Comment by DirtyF** [comment]
seems related to `--livereload` dependencies. 🤔

File: lib/jekyll/commands/serve/websockets.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def dispatch(data)
           # WebSockets requests will have a Connection: Upgrade header
           if parser.http_method != "GET" || parser.upgrade?
             super
-          elsif parser.request_url =~ %r!^\/livereload.js!
+          elsif %r!^\/livereload.js!.match?(parser.request_url)
```
[the remark this answers is not in the document]
Comment (ashmaroli) [reply lib/jekyll/commands/serve/websockets.rb]: I didn't block the PR. Just left a benign suggestion which @DirtyF wisely chose to ignore. So, I don't see what the issue here is..

File: lib/jekyll/commands/serve/servlet.rb [hunk 2]
```diff
@@ -186,7 +186,7 @@ def validate_and_ensure_charset(_req, res)
           key = res.header.keys.grep(%r!content-type!i).first
           typ = res.header[key]
 
-          unless typ =~ %r!;\s*charset=!
+          unless %r!;\s*charset=!.match?(typ)
```
Comment (mattr-) [inline lib/jekyll/commands/serve/servlet.rb]: These don't seem strictly necessary. Can we minimize the diff in this PR by not including these changes here?

**Comment by ashmaroli** [comment]
I've currently configured Travis to let JRuby builds to fail since the failures cannot be "fixed" from our end.
Once the JRuby builds start passing (with a future JRuby version), we can remove the `allow_failures` configuration.

**Comment by mattr-** [comment]
Why can't the JRuby builds be fixed?

**Comment by ashmaroli** [comment]
> Why can't the JRuby builds be fixed?

@mattr- Based on https://travis-ci.org/jekyll/jekyll/jobs/476924353 the build fails due to a method missing in a second-order dependency of Jekyll (because JRuby has issues with Refinements).
One way to "fix the build" is to workaround the issue by "skipping" the failing test entirely.

But I feel the better way would be to let the test fail without affecting the overall build status.

**Comment by mattr-** [comment]
That's unfortunate. I'd rather skip the specific test on JRuby, keep that test suite as required, and see other failures from that platform as they happen, so that we can fix them if they are in our code vs. a dependency.

**Comment by ashmaroli** [comment]
> and see other failures from that platform as they happen

Valid point! I'll make the changes. :+1:

**Comment by ashmaroli** [comment]
Note: This will require updating Travis config to JRuby - 9.2.7.0 (currently under development)
The reason for the failing test here has been purportedly fixed upstream via  &nbsp; https://github.com/jruby/jruby/commit/e509934e078c72ce033ad0d96116960748d48afd

**Comment by DirtyF** [comment]
@jekyllbot: merge +major

**Pull request merged**